app/controllers/api/v1/post_tags_controller.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    # Controller class that is responsible for handling post tags requests.
+    class PostTagsController < ApplicationController
+      include DeviseTokenAuth::Concerns::SetUserByToken
+      include Pundit
+
+      before_action :authenticate_user!
+      before_action :set_post_tag, only: %i[destroy]
+      after_action :verify_authorized
+
+      # POST /post_tags or /post_tags.json
+      def create
+        if post_tag_params.blank?
+          render json: { error: 'There was no post tag data passed in so your user post tag could not be saved.' },
+                 status: :unprocessable_entity
+        else
+          @post_tag = PostTag.new(post_tag_params)
+          authorize @post_tag
+          if @post_tag.save
```

Comment:
```
#post_contorller
post = Post.find_by(id: ...)

post.tags.create(tag_id: ...)

#path
POST /posts/:id/add_tag 

#routes
resources :posts do
  post /add_tag, to: 'posts#add_tag'
end
```

Reply:
#post_contorller
post = Post.find_by(id: ...)
tag = Tag.find_or_create_by(text: ...)

post.tags.create(tag_id: tag.id)

#path
POST /posts/:id/add_tag 

#routes
resources :posts do
  post /add_tag, to: 'posts#add_tag'
end